File: Models/Dependencies.js
// File = robjam1990/Psychosis/Models/Dependencies.js
// Initialize dependencies
const express = require('express');
const bodyParser = require('body-parser');
const app = express();
const PORT = 5000;

// Serve MLflow model
app.post('/invocations', bodyParser.json(), (req, res) => {
    // Handle different types of input data formats

    // Serialized pandas DataFrame format
    if (req.body.dataframe_records) {
        const data = req.body.dataframe_records;
        // Process data
        console.log(data);
        res.send(data); // Sending back processed data as response (dummy response)
    }
    // Split serialized pandas DataFrame format
    else if (req.body.dataframe_split) {
        const { columns, index, data } = req.body.dataframe_split;
        // Process data
        console.log(columns, index, data);
        res.send(data); // Sending back processed data as response (dummy response)
    }
    // List format for processing
    else if (req.body.inputs) {
        const inputs = req.body.inputs;
        // Process data
        console.log(inputs);
        res.send(inputs); // Sending back processed data as response (dummy response)
    }
    // Tensor data instances for processing
    else if (req.body.instances) {
        const instances = req.body.instances;
        // Process data
        console.log(instances);
        res.send(instances); // Sending back processed data as response (dummy response)
    } else {
        res.status(400).send('Invalid request');
    }
});

// Start server
app.listen(PORT, () => {
    console.log(`Server is running on port ${PORT}`);
});

File: Models/Dependencies.py
# File: robjam1990/Psychosis/Models/Dependencies.py
from flask import Flask, request, jsonify

app = Flask(__name__)
PORT = 5000

# Serve MLflow model
@app.route('/invocations', methods=['POST'])
def invocations():
    req_data = request.get_json()

    # Handle different types of input data formats

    # Serialized pandas DataFrame format
    if 'dataframe_records' in req_data:
        data = req_data['dataframe_records']
        # Process data
        print(data)
        return jsonify(data)  # Sending back processed data as response (dummy response)

    # Split serialized pandas DataFrame format
    elif 'dataframe_split' in req_data:
        columns = req_data['dataframe_split']['columns']
        index = req_data['dataframe_split']['index']
        data = req_data['dataframe_split']['data']
        # Process data
        print(columns, index, data)
        return jsonify(data)  # Sending back processed data as response (dummy response)

    # List format for processing
    elif 'inputs' in req_data:
        inputs = req_data['inputs']
        # Process data
        print(inputs)
        return jsonify(inputs)  # Sending back processed data as response (dummy response)

    # Tensor data instances for processing
    elif 'instances' in req_data:
        instances = req_data['instances']
        # Process data
        print(instances)
        return jsonify(instances)  # Sending back processed data as response (dummy response)

    else:
        return 'Invalid request', 400

# Start server
if __name__ == '__main__':
    app.run(port=PORT)
